content/news/XXXX-XX-XX-gsoc-final-blog-fatih-emre-yildiz.md

```diff
@@ -0,0 +1,395 @@
+
+The best use case was to change the `Import Metadata From Musicbrainz` window to be suitable for cover art fetching. That was solving all the possible issues mentioned.  
+
+Right now, users can select between the lowest quality to highest quality, when they press apply on the tag fetcher cover art is downloaded and updated for that track.
+
+Related files:
+
+[coverartarchivelinkstask.cpp](https://github.com/mixxxdj/mixxx/blob/18ed28d9061feb19b58aa4db1730f347622686a5/src/musicbrainz/web/coverartarchivelinkstask.cpp)
+
+[coverartarchiveimagetask.cpp](https://github.com/mixxxdj/mixxx/blob/18ed28d9061feb19b58aa4db1730f347622686a5/src/musicbrainz/web/coverartarchiveimagetask.cpp)
+
+
+*Cover Art Fetcher*
+![Cover Art Fetcher]({static}/images/news/coverartfetcher.png)
+
+##### *Track Suggestion Feature*
+
+This is the main part of the project. In my opinion Track Suggestion is a cool feature for a DJ app. Getting a hint for the next track at right time would be so precious for a DJ. But the possibility of the finding correct track has also the same amount of difficulty. 
+
+To find similar tracks, we have used Last.fm Thanks to their [API](https://www.last.fm/api) we can get track suggestions by just providing the Artist and Title. What is changed is basically on the library sidebar, there is a new feature called "Track Suggestion" which can be turned on or turned off on the preferences page. There are five different sub-menus located under this feature. Four of them belong to the decks and one of them belongs to the user's choice. When a track is placed on a deck, the correlated sub-menu changes to the `Artist | Track Title`. If the user clicks on that menu a new library is populated. If the suggestions are fetched before the user will be retrieved from the database, If it is not, the user will see a button that says `Load Track Suggestions For Artist | Track Title`. After this button is pressed, we send the request and get the response cached in the database.
+
+The new library is an External Library. The located tracks are not available for to play, because they are only text retrieved from Last.fm. But the tracks listed in that library can give an idea to the DJ about what to play next. DJ can see a title or an artist and this can help about picking the next song.
+
+There is a problem with the Last.fm service, when the track or artist name is misspelled the suggestions can not be fetched. Also for some tracks, there is no suggestion available on the Last.fm servers. To fix this issue, there was another service implementation that was in process, to get the same bpm or key.
```

Comment:
How does key and bpm relate to a missing suggestion?

Reply:
That was related with the getsongbpm, but that shouldn't be on here I guess. But a future reference can be added